iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -744,8 +745,17 @@ private String collectColumnAndReplaceDummyValues(ExprNodeDesc node, String foun
     return column;
   }
 
-  private void fallbackToNonVectorizedModeForV2(Properties tableProps) {
-    if ("2".equals(tableProps.get(TableProperties.FORMAT_VERSION))) {
+  /**
+   * If any of the following checks is true we fall back to non vectorized mode:
+   * <ul>
+   *   <li>iceberg format-version is "2"</li>
+   *   <li>fileformat is set to avro</li>
+   * </ul>
+   * @param tableProps table properties, must be not null
+   */
+  private void fallbackToNonVectorizedModeBasedOnProperties(Properties tableProps) {
+    if ("2".equals(tableProps.get(TableProperties.FORMAT_VERSION)) ||
+        FileFormat.AVRO.name().equalsIgnoreCase((String) tableProps.get(TableProperties.DEFAULT_FILE_FORMAT))) {
```

Comment:
nit: you can use `tablePros.getProperty()` instead of casting

Reply:
yepp, fixed it.